Compiler/BackEnd/Matching.mo

```diff
@@ -5666,7 +5666,7 @@ algorithm
       arrayUpdate(m,idx,row);
       //update mt
       for varIdx in varIdxs loop
-        row := arrayGet(m,varIdx);
+        row := arrayGet(mt,varIdx);
```

Comment:
@sjoelund Is this fix correct? From my understanding mt should be updated here, but maybe it is the other way round!?

Reply:
@vwaurich Can you have a look please?

Reply:
I'm not sure. I guess it's reasonable to get the value and update it instead of getting some different value...